src/proxy_configuration.js

```diff
@@ -210,7 +222,7 @@ export class ProxyConfiguration {
         if (sessionId) this._validateSessionArgumentStructure(sessionId);
         const url = this.newUrl(sessionId);
 
-        const { groups, countryCode, password, port, hostname } = this.proxyUrls ? new URL(url) : this;
+        const { groups, countryCode, password, port, hostname } = this.proxyUrls || this.newUrlFunction ? new URL(url) : this;
```

Comment:
I would use `this.usesApifyProxy` that would either be a function or a property set in constructor. You keep using the same checks over and over and it's not clear that `this.proxyUrls || this.newUrlFunction` actually means "The ProxyConfiguration uses Apify Proxy"

Reply:
Changed.